### PR TITLE
Update docs for `concat` in case `axis < 0`

### LINKED ISSUE #15905: Documentation does not explain the utility of -1 as value for the axis parameter of the tf.concat method
### System information

- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS X, version 10.13.2
- **TensorFlow installed from (source or binary)**: Binary (pip)
- **TensorFlow version (use command below)**: v1.3.0-rc1-5211-gab0fcac 1.5.0-dev20171126
- **Python version**: Python 3.5.0
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**:  N/A
- **CUDA/cuDNN version**:  N/A
- **GPU model and memory**:  N/A
- **Exact command to reproduce**: Just run a Python script with the code I am sharing with you

### Describe the problem

It apparently concatenates along the last axis. See the following example:  

```
import tensorflow as tf

t1 = [[[1, 2], [2, 3]], [[4, 4], [5, 3]]]
t2 = [[[7, 4], [8, 4]], [[2, 10], [15, 11]]]

with tf.Session() as sess:
    result = sess.run(tf.concat([t1, t2], -1))
    print(result)
```

which produces

```
[[[ 1  2  7  4]
  [ 2  3  8  4]]

 [[ 4  4  2 10]
  [ 5  3 15 11]]]
``` 

The following documention does not seem to explain this use case:

- https://www.tensorflow.org/api_docs/python/tf/concat
- https://www.tensorflow.org/versions/r1.5/api_docs/python/tf/concat

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #15905 where the documentation does not cover the case of `axis < 0` for `tf.concat`.

This fix fixes #15905.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/python/ops/array_ops.py [hunk 1]
```diff
@@ -1107,7 +1107,8 @@ def concat(values, axis, name="concat"):
   Args:
     values: A list of `Tensor` objects or a single `Tensor`.
     axis: 0-D `int32` `Tensor`.  Dimension along which to concatenate. Must be
-      in the range `[-rank(values), rank(values))`.
+      in the range `[-rank(values), rank(values))`. If a negative number for
```
Comment (caisq) [inline tensorflow/python/ops/array_ops.py]: I would reword this to emphasize that this is the same as indexing in Python.

**Review by caisq — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/array_ops.py (hunk 1 above)
Comment (yongtang) [reply tensorflow/python/ops/array_ops.py]: Thanks @caisq. The docstring has been updated.

**Review by caisq — APPROVED** [review]

**Comment by caisq** [comment]
@tensorflow-jenkins test this please
